### PR TITLE
Japan: substitute holidays and citizens' holidays calculation

### PR DESCRIPTION
**Pull request by KJhellico**

### CODE REVIEW

**Comment by KJhellico** [comment]
I'm not sure, but maybe "substitute holidays" is `observed`? Or it's just holidays of special type?

File: holidays/countries/japan.py [hunk 1]
```diff
@@ -143,166 +143,22 @@ def _populate(self, year):
...
-        )
-        for holiday in table:
-            month = holiday[0]
-            day = holiday[1]
-            years = holiday[2]
-            if year in years:
-                self[date(year, month, day)] = "振替休日"
+        # When a national holiday falls on Sunday, next working day
+        # shall become a public holiday (振替休日) - substitute holidays
+        for dt in sorted(list(self.keys())):
```
Comment (arkid15r) [inline holidays/countries/japan.py:148]: I'm not sure if you need `sorted()` here and below. But if you do, the `for dt in sorted(self.keys()):` would work just fine.

**Review by arkid15r — COMMENTED** [review]
>I'm not sure, but maybe "substitute holidays" is observed? Or it's just holidays of special type?

My understanding is that "substitute holiday" is a direct translation of "振替休日". And the meaning of that seems to be equal to "observed holiday" term.

🇯🇵  LGTM!

File: holidays/countries/japan.py (hunk 1 above)
Comment (KJhellico) [reply holidays/countries/japan.py]: Yes, `sorted()` isn't needed here. Thank you!

**Comment by KJhellico** [comment]
> My understanding is that "substitute holiday" is a direct translation of "振替休日". And the meaning of that seems to be equal to "observed holiday" term.

I think so too. So make it `self.observed` depended now.

**Comment by arkid15r** [comment]
Great refactoring 👍, thank you!

**Pull request merged**